### PR TITLE
'url' key is the GET data parameter and  'data' key stands for the POST o…

### PR DESCRIPTION
**Pull request by b23prodtm**
…nes for testAction()

Correct information was found in the Controller::requestAction() documentation.

### CODE REVIEW

File: en/development/testing.rst [hunk 1]
```diff
@@ -886,7 +886,7 @@ method key::
                 'body' => 'Secret sauce'
             )
         );
-        $this->testAction('/posts/add', array('data' => $data, 'method' => 'get'));
+        $this->testAction('/posts/add', array('url' => $data, 'method' => 'get'));
```
Comment (markstory) [inline en/development/testing.rst]: This isn't for requestAction though.

Comment (b23prodtm) [reply en/development/testing.rst]: But it's what caused the issue on my side, and I found the solution in requestAction docs. The key 'url' needs to replace 'data' in testAction() as well.

Comment (b23prodtm) [reply en/development/testing.rst]: Got this : 
```
POST and GET data can be simulated in requestAction. Use `$extra['url']` for GET data. The `$extra['data']` parameter allows POST data simulation.
```
That worked well in my tested controller, which wouldn't send GET variables before as the 'data' key.

Comment (markstory) [reply en/development/testing.rst]: Ok. It has been a while since I have read testAction(). I will read through it before merging this.

Comment (markstory) [reply en/development/testing.rst]: This https://github.com/cakephp/cakephp/blob/e889535e4172feeeac77268b0f072823a34199cd/lib/Cake/TestSuite/ControllerTestCase.php#L242-L250 makes me think that the current documentation is correct. Are you sure you're not confusing `testAction()` which is used in test cases, and `requestAction()` which is used for sub-requests? You're completely right about how `requestAction()` works with `$extra['url']`.

Comment (b23prodtm) [reply en/development/testing.rst]: Uh nop,  I was mistaken.  I investigated further in the source code.
I was reading my GET values with filter_input(INPUT_GET, $var) , whihc doesn't allow $_GET global be altered in some way.. **ControllerTestCase.php** alters $_GET in _testAction() which is not read by **php's filter_input()**.

What was the cause ? I did manually change to requestAction in another file, which was conflicting with the values in the test file and obviously altering my data.

**Pull request closed without merging**